### ISSUE TITLE
Plots of external data in QC

### ISSUE THREAD

**Issue by squaregoldfish** [BjerknesClimateDataCentre/QuinCe#435]
In QC, allow data to be plotted against external data, e.g. for pressure plot against NCEP/NCAR pressures. Can be a climatology, the data from the same period in the previous year, or something else - research what would be suitable.

This can be done in the plots as a simple second series, or the fields could be plotted as raster grids on the maps.

**Comment by squaregoldfish** [comment]
SOCAT, SST, Salinity. Previous instrument data from within QuinCe (see #498)?

**Comment by squaregoldfish** [comment]
Also background for maps?

**Comment by squaregoldfish** [comment]
Also use for Auto QC - compare dataset data with remote dataset and have a deviation threshold for each position/time.

**Comment by squaregoldfish** [comment]
Data needs to be retrieved on demand, but also cached for speed. At the same time each dataset only needs a subset of data, so we need to cache subsets. Which might overlap etc.

Also need primary/fallback datasets per parameter. Eg use delayed mode data where available, otherwise NRT data. (OISST publishes these.)

If data is on an ERDDAP it'll probably help, since we can easily do subsetting queries and get netCDF back.

How long do we cache data for? Throw away once automatic export is complete? As long as it's not used elsewhere... Keep one cache per dataset? Retrieve data on extraction, once the bounds (temporal and spatial) have been established.

**Comment by squaregoldfish** [comment]
Water depth and/or distance from coast

**Comment by squaregoldfish** [comment]
Anna Canning suggested being able to manually upload a data file against a dataset for plotting/comparison. This would be very hard due to defining the format, and may be split out into a separate issue.
